### PR TITLE
Add: Development and build/ release guide and edit button to docs theme

### PR DESCRIPTION
**Pull request by lwasser**
This PR does a few things

✅ adds a development guide to overview our dev processes
✅ Adds a build and release guide for build and release workflows
✅ adds an edit button to the furo theme for easy small edits that contributors might want to make to the docs
✅ creates a PR template for releases to make the process a bit simpler for maintainers 

@jsamoocha we now have two sets of tests and i just am hoping you can look at that section closely and suggest changes. i suspect we're moving away from some of the tests that hit the API directly. But i haven't dug in deeply enough to that part of the tests yet to understand the overlap of what is mocked vs what requires direct API access.  

Also i wonder if you have the ability to create a release. It might be worth at some point releasing a patch to ensure you can do that too!

Finally, when you have time, can you kindly try to login to codecov.io with your github account? i was wondering if that is one way that might allow you to see the line by line reports provided both on pr's and the master branch of the repository. (obviously that last part isn't related to this pr i just want to ensure that you have sufficient access to things!

### CODE REVIEW

**Comment by lwasser** [comment]
It's super painful to read the docs output and see what is failing. let me see if i can print out the failures to improve this. moving to draft

**Comment by lwasser** [comment]
Ok @jsamoocha ready for review. this is another BIG pr - sorry. I printed the docs output in another step of the build. 

note: 

```
contributing/how-to-contribute.md:16: [broken] https://stravalib.readthedocs.io/contributing/development-guide.html: 404 Client Error: Not Found for url: https://stravalib.readthedocs.io/contributing/development-guide.html 
```

The page that can't be found is in this PR. And it's not a relative link because it's a link from our contributing.md file to the documentation. So it will fail until we merge this PR. IF we are concerned with a red x on a merge, i can REMOVE the link from the build and add it in a separate tiny pr (with no changelog associated with it). Just let me know if you'd prefer that OR if you're ok with this one point of failure in this PR we can leave as is. (unless you find other build failures that i missed as you normally do 😄 )

**Review by jsamoocha — COMMENTED** [review]
Excellent work; it will lower the threshold to contribute to this package!

Besides the specific inline comments, I noticed that not all headings are capitalized. Did you intend only to capitalize H1s? If so, that seems to be done consistently, and nothing needs to change IMO.

**Comment by jsamoocha** [comment]
> Also i wonder if you have the ability to create a release. It might be worth at some point releasing a patch to ensure you can do that too!

After we merge this, I'll try to make a 1.0.1 release using the template you made.

> Finally, when you have time, can you kindly try to login to codecov.io with your github account?

I could see the line-by-line coverage on master, even while being logged out of github.

File: CONTRIBUTING.md
Comment (lwasser) [reply CONTRIBUTING.md:108]: ah let me look at this locally. i added the text coverage stuff at the end of a longer editing session so it could be redundant!  thanks!

File: docs/conf.py
Comment (lwasser) [reply docs/conf.py:93]: it should ask someone without edit access to the repo to fork the repo if it's not forked. [try it here in the theme documentation itself so you can see how it works.](https://pradyunsg.me/furo/quickstart/) I believe that because you do have write access it will allow you to edit directly. But only you, myself, yihong and hans have write access so i think it should be ok as long as none of us go crazy trying to edit from the edit button :) I can also add protections to the master branch to prevent that from happening by mistake - it will just ask us if we do intend to push directly to master IF we want that.

File: docs/contributing/development-guide.md
Comment (lwasser) [reply docs/contributing/development-guide.md:176]: that would be great. If the test section gets too long we can also create a new page for it in the documentation. I did that with the release content. i didn't want the development guide to get huge and figured some things like releases would only be done by us. The test suite could also be something that most people won't touch. 
Just a thought. in the meantime i'll work on this comment locally later today !

File: docs/contributing/development-guide.md
Comment (lwasser) [inline docs/contributing/development-guide.md:202]: NOTE TO SELF: update this and remove tests from the file. That was the pr i merged this morning.

File: docs/contributing/development-guide.md
Comment (lwasser) [reply docs/contributing/development-guide.md:263]: ok just in case it's nice for someone else to have access :) do you have a readthedocs account? if not [can you please signup here](https://readthedocs.org/accounts/signup/) you can use your github account. That give me your username and i'll add you. Fixes will be  rare and i can  handle it i just worry if i'm offline and something comes up it's nice to have others with access. hans has access too.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (jsamoocha) [inline .github/PULL_REQUEST_TEMPLATE.md:5]:
```suggestion
* If this PR closes an open issue, please add to the very top of this pull request (above the title) the words 
```

File: docs/contributing/development-guide.md
Comment (jsamoocha) [reply docs/contributing/development-guide.md:263]: My username at readthedocs is `jsamoocha` (but not linked to my Github account (yet)).

**Review by jsamoocha — APPROVED** [review]
Besides one more small typo, this looks good to merge. Thanks!

File: .github/PULL_REQUEST_TEMPLATE/release-pull-request-template.md
Comment (lwasser) [reply .github/PULL_REQUEST_TEMPLATE/release-pull-request-template.md:1]: Ok no worries... so i'm super glad you asked this because it actually works a bit differently than i thought it worked. 

You can make [issue templates ](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository)which are nice because they allow someone submitting a new issue to chose the type of issue. You can even have github automatically label issues by type. and in the template you ask people for specific information like for us we may want to see the users environment list, version of python, etc for all bugs reported. 

Pull request templates are a little less nice. We can have a nice default template (which i just added i'll ping you on the file). that text will then appear on any new PR opened. we can ask users to ensure they read the contributing guide for instance, and opened an issue first. I asked them to add closes #issue-number-here at the top because with that text GitHub will close their issue when we merge making maintenance a bit easier! 

Now the release template let's see if we like this. It turns out that we need to craft a specific URL to grab that release PR template. IT will just be you and I making releases i think for the most part. So let's see if we like this. I added a LINK at the bottom of the normal PR template to

It will look like this:
(i added release-1.1 as the branch name you'd have to edit the url to make sure the branch is right). I haven't tested 
https://github.com/stravalib/stravalib/compare/master...release-1.1?template=release-pull-request-template.md

I think i've set this up so you can see how it works in my fork. i'll add you as a maintainer to my fork and you can see. 

Try the link below 

https://github.com/lwasser/stravalib/compare/master...test-br?template=release-pull-request-template.md 

Then try to open a normal pr on that branch and notice when you hit "create pull request" it populates text for you in the pr text box. 

Let me know if that makes sense!

Comment (lwasser) [reply .github/PULL_REQUEST_TEMPLATE/release-pull-request-template.md:1]: issues are much cleaner. if the PR pull request is more effort than it's worth i can just create some template text in the release guide too that we can copy over with the check list.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (lwasser) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: @jsamoocha this is the pr template that everyone will now see for every pr opened.

File: .github/PULL_REQUEST_TEMPLATE/release-pull-request-template.md
Comment (lwasser) [inline .github/PULL_REQUEST_TEMPLATE/release-pull-request-template.md:3]: This is the template that is specific to a release and requires a URL to access

Comment (lwasser) [reply .github/PULL_REQUEST_TEMPLATE/release-pull-request-template.md:3]: instructions are in the main PR template above about accessing it

File: CONTRIBUTING.md
Comment (lwasser) [reply CONTRIBUTING.md:108]: it is. i removed it!

File: docs/contributing/development-guide.md
Comment (lwasser) [reply docs/contributing/development-guide.md:202]: ok fixed this locally will push in a minute.

**Comment by lwasser** [comment]
Ok - i *think* i've addressed everything. **Ready for another review.** 

I think we may need to play with the PR templates. the main one is a good idea and we can edit it - note it's a new file to review in this pr. i'm curious how the release template will work. if we find we don't use it we can just delete it and i can make a easy to copy markdown block in our docs.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (jsamoocha) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: Ok, looks good! I didn't see this one on your fork, though. Through the link you gave (when removing the `?template=release-pull-request-template.md` param), I saw a template. But it looked like some generic placeholder. 

If it is not working yet and will take more time to fix, maybe we should add the templates in another PR? You clearly described the release process in the docs, and we can always use that as a checklist.

File: CONTRIBUTING.md
Comment (jsamoocha) [inline CONTRIBUTING.md:115]:
```suggestion
We use `codecov` implemented through the `pytest-cov` extension to sphinx to 
track `stravalib`'s test % coverage. When you submit a pull request, you will 
see how that pull request affects our package's total test coverage.   
```

**Review by jsamoocha — COMMENTED** [review]
LGTM. I checked the release template on your fork using the extra `?template=release-pull-request-template.md` request parameter. However, when removing this parameter, I saw a default PR template that was different from the one I expected (i.e., the one edited in your recent commits). But maybe my expectation is wrong and what I see is the result of some pending branch merging that still needs to happen at your end.

File: .github/PULL_REQUEST_TEMPLATE/release-pull-request-template.md
Comment (jsamoocha) [reply .github/PULL_REQUEST_TEMPLATE/release-pull-request-template.md:1]: Cool, thanks for the explanation! I can see how the release template works and will not be there by default to confuse authors of regular PRs 👍.

File: docs/contributing/development-guide.md
Comment (lwasser) [reply docs/contributing/development-guide.md:176]: opps i missed this is think. will add it tonight. i'll go through this entire PR tonight to make sure i've caught all of the comments.

**Review by jsamoocha — APPROVED** [review]
Great, let's merge this!

**Comment by lwasser** [comment]
thank you @jsamoocha for the great reviews on this!! Merging now!

**Pull request merged**
